mimic/rest/auth_api.py

```diff
@@ -44,41 +45,79 @@ def get_token_and_service_catalog(self, request):
 
         tenant_id = (content['auth'].get('tenantName', None) or
                      content['auth'].get('tenantId', None))
+
+        def format_response(callable_returning_session,
```

Comment:
This function is starting to push the boundaries of being too long.  Looking at it, I don't see any obvious way to break it into reasonable smaller pieces, but it's something to think about for future changes here.